backbone/src/main/java/org/researchstack/backbone/ui/step/fragments/BaseStepFragment.kt

```diff
@@ -77,6 +67,18 @@ internal open class BaseStepFragment(@LayoutRes contentLayoutId: Int) : Fragment
                 Log.d("BaseStepFragment", "WARNING: Unknown stepView type, cannot initialize layout")
             }
         }
+
+        if (stepView is StepLayout) {
```

Comment:
Is there any reason why you're not doing this in the `when` above?

There's a clause `is StepLayout -> {` so you could do it in there, instead of checking here again. This would allow you to leave the `when (...)` line the way it was before, you no longer need the smart-cast to StepLayout here.

Reply:
@Gryzor It's because I will have to add the code in both blocks 
            is SurveyStepLayout -> { // this is the class where implementing StepLayout
            is StepLayout -> { // this is the for IC, BC,AC RelativeLayout where implementing StepLayout 
so i will add same code for both